### PR TITLE
Fix mariadb permissions

### PR DESCRIPTION
**Pull request by alexjh**

### CODE REVIEW

File: stable/mariadb/templates/master-statefulset.yaml [hunk 1]
```diff
@@ -50,6 +50,18 @@ spec:
         - name: {{ . }}
       {{- end}}
       {{- end }}
+      {{- if .Values.master.persistence.enabled }}
+      initContainers:
+      - name: init-{{ template "mariadb.fullname" . }}
+        image: "{{ template "mariadb.image" . }}"
+        command: ["sh", "-c", "chmod 0777 /bitnami/mariadb"]
```
Comment (mook-as) [inline stable/mariadb/templates/master-statefulset.yaml]: Does this need the full `777`, or just be owned by the right user?

File: stable/mariadb/templates/slave-statefulset.yaml [hunk 2]
```diff
@@ -51,6 +51,18 @@ spec:
         - name: {{ . }}
       {{- end}}
       {{- end }}
+      {{- if .Values.slave.persistence.enabled }}
+      initContainers:
+      - name: init-{{ template "mariadb.fullname" . }}
+        image: "{{ template "mariadb.image" . }}"
+        command: ["sh", "-c", "chmod 0777 /bitnami/mariadb"]
```
Comment (mook-as) [inline stable/mariadb/templates/slave-statefulset.yaml]: Same as above.

**Review by mook-as — COMMENTED** [review]
Not sure upstream would want to take this…

Are we seeing this because we're using a badly-setup NFS (or hostpath?) volume mount?

File: stable/mariadb/templates/master-statefulset.yaml (hunk 1 above)
Comment (andreas-kupries) [reply stable/mariadb/templates/master-statefulset.yaml]: Taking over from alex, I can't say.
Who would be the correct user ?

Comment (andreas-kupries) [reply stable/mariadb/templates/master-statefulset.yaml]: Also note that https://github.com/SUSE/charts/pull/6 does the equivalent change for mongodb.

Comment (mook-as) [reply stable/mariadb/templates/master-statefulset.yaml]: Looking at the relevant [dockerfile](https://github.com/bitnami/bitnami-docker-mariadb/blob/96d18812ab8705b5197f4258a73ab8616b6839cf/10.1/debian-9/Dockerfile#L37), looks like it's `1001`?

Comment (andreas-kupries) [reply stable/mariadb/templates/master-statefulset.yaml]: Yeah. Looks to be it. Will make the change. Currently not set up for testing it however. We will see.

Comment (andreas-kupries) [reply stable/mariadb/templates/master-statefulset.yaml]: Change made with commit dd2449f18718ceb81f2dab88ab6fad329175bc11

File: stable/mariadb/templates/slave-statefulset.yaml (hunk 2 above)
Comment (andreas-kupries) [reply stable/mariadb/templates/slave-statefulset.yaml]: Change made with commit dd2449f18718ceb81f2dab88ab6fad329175bc11

File: stable/mariadb/templates/master-statefulset.yaml (hunk 1 above)
Comment (andreas-kupries) [reply stable/mariadb/templates/master-statefulset.yaml]: See also commit  f32c426dc47c3e3b5fe7228974cce83cb6cccf2e over on https://github.com/SUSE/charts/pull/6

File: stable/mariadb/templates/master-statefulset.yaml [hunk 3]
```diff
@@ -50,6 +50,18 @@ spec:
         - name: {{ . }}
       {{- end}}
       {{- end }}
+      {{- if .Values.master.persistence.enabled }}
+      initContainers:
+      - name: init-{{ template "mariadb.fullname" . }}
+        image: "{{ template "mariadb.image" . }}"
+        command: ["sh", "-c", "chown 1001 /bitnami/mariadb"]
```
Comment (mook-as) [inline stable/mariadb/templates/master-statefulset.yaml:57]: Dumb question: why `sh -c`? (I know you didn't write the code)

Comment (andreas-kupries) [reply stable/mariadb/templates/master-statefulset.yaml:57]: I suspect he based his code on the examples at https://kubernetes.io/docs/concepts/workloads/pods/init-containers/

**Review by mook-as — APPROVED** [review]
Don't care enough to make you fix things

File: stable/mariadb/templates/master-statefulset.yaml (hunk 3 above)
Comment (andreas-kupries) [reply stable/mariadb/templates/master-statefulset.yaml:57]: The example at https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-initialization/ is without `sh -c`. So consistent. I would stay with the current code.

**Pull request merged**